lxd/ubuntu-pro/watcher.go

```diff
@@ -0,0 +1,252 @@
+package pro
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+// GuestAttachSetting represents ubuntu-pro settings for guest attachment availability.
+type GuestAttachSetting string
+
+const (
+	// GuestAttachSettingOff indicates that guest attachment is never available. The value of `GET /1.0/ubuntu-pro` over
+	// devlxd should always be "off".
+	GuestAttachSettingOff GuestAttachSetting = "off"
+
+	// GuestAttachSettingAvailable indicates that guest attachment is available and can be opted into via guest config.
+	// The value of `GET /1.0/ubuntu-pro` over devlxd should return the value of the `ubuntu_pro.guest_attach`
+	// configuration of the instance.
+	GuestAttachSettingAvailable GuestAttachSetting = "available"
+
+	// GuestAttachSettingOn indicates that guest attachment is on and can be opted into via guest config. The value of
+	// `GET /1.0/ubuntu-pro` over devlxd should return the value of the `ubuntu_pro.guest_attach` configuration of the
+	// instance.
+	//
+	// NOTE: This setting currently behaves in the same way as GuestAttachSettingAvailable, but we will use this setting
```

Comment:
This isn't correct. The pro client inside the guest will trigger auto attach for `on` but not for `available` (that would require manual `pro auto-attach`).

Reply:
I'm not sure we should include the text about the future work as that is not 100% agreed yet.